src/tools/pkcs11-tool.c

```diff
@@ -4895,26 +4910,31 @@ derive_ec_key(CK_SESSION_HANDLE session, CK_OBJECT_HANDLE key, CK_MECHANISM_TYPE
 #endif
 
 	if (!pkey)
-		util_fatal("Cannot read EC key from %s", opt_input);
+		util_fatal("Cannot read peer EC key from %s", opt_input);
+
+	key_id = EVP_PKEY_id(pkey);
+
+	switch (key_id) {
+	case EVP_PKEY_EC: /* CKK_EC*/
 
 #if OPENSSL_VERSION_NUMBER < 0x30000000L
-	eckey = EVP_PKEY_get0_EC_KEY(pkey);
-	ecpoint = EC_KEY_get0_public_key(eckey);
-	ecgroup = EC_KEY_get0_group(eckey);
+		eckey = EVP_PKEY_get0_EC_KEY(pkey);
+		ecpoint = EC_KEY_get0_public_key(eckey);
+		ecgroup = EC_KEY_get0_group(eckey);
 
-	if (!ecpoint || !ecgroup)
-		util_fatal("Failed to parse other EC key from %s", opt_input);
+		if (!ecpoint || !ecgroup)
+			util_fatal("Failed to parse peer EC key from %s", opt_input);
 #else
-	if (EVP_PKEY_get_group_name(pkey, name, sizeof(name), &len) != 1
-	 || (nid = OBJ_txt2nid(name)) == NID_undef
-	 || (ecgroup = EC_GROUP_new_by_curve_name(nid)) == NULL)
-		util_fatal("Failed to parse other EC key from %s", opt_input);
+		if (EVP_PKEY_get_group_name(pkey, name, sizeof(name), &len) != 1 || (nid = OBJ_txt2nid(name)) == NID_undef || (ecgroup = EC_GROUP_new_by_curve_name(nid)) == NULL)
```

Comment:
This line looks too long. Please, split it. With this splitting, the clang should not complain.
```suggestion
		if (EVP_PKEY_get_group_name(pkey, name, sizeof(name), &len) != 1 ||
				(nid = OBJ_txt2nid(name)) == NID_undef || 
				(ecgroup = EC_GROUP_new_by_curve_name(nid)) == NULL)
```

Reply:
not resolved